lib/diagnostic.rb

```diff
@@ -26,26 +26,36 @@
 # be writable.
 
 ##
-# your response here
+class Person
+  attr_reader :name, :age
+  attr_writer :name, :location
+end
 ##
 
 ##
 # Instatiate a new Person object using arguments "Dave", 32, and "Ohio".
 # Then, in the next line, change this Person's location property to
 # "Somerville". Finally, assign the modified Person to `Response.dave` below.
-
+def initialize (Dave,32, Ohio)
+ @location = Somerville
+ @name = Dave
+ @age = 32
 ## replace nil with your response, then continue your work on the next line
-dave = nil
+dave = Person.new
 Response.dave = dave
 ##
-
+end
 ##
 # Create another class called Developer that inherits from Person.
 # Give it a new instance method called 'solve_problems',
 # which returns the string "think think think".
 
 ##
-# your response here
+class Developer < Person
+  def instance
```

Comment:
This defines an instance method called `instance` that sets an instance variable called `@solve_problems` to the string `'think think think'`.